pkg/cmd/kind/load/docker-image/docker-image.go

```diff
@@ -121,38 +126,40 @@ func runE(logger log.Logger, flags *flagpole, args []string) error {
 
 	// pick only the nodes that don't have the image
 	selectedNodes := []nodes.Node{}
-	for _, node := range candidateNodes {
-		id, err := nodeutils.ImageID(node, imageName)
-		if err != nil || id != imageID {
-			selectedNodes = append(selectedNodes, node)
-			logger.V(0).Infof("Image: %q with ID %q not yet present on node %q, loading...", imageName, imageID, node.String())
+	fns := []func() error{}
+	for i, imageName := range imageNames {
+		imageID := imageIDs[i]
+		for _, node := range candidateNodes {
+			id, err := nodeutils.ImageID(node, imageName)
+			if err != nil || id != imageID {
+				selectedNodes = append(selectedNodes, node)
+				logger.V(0).Infof("Image: %q with ID %q not yet present on node %q, loading...", imageName, imageID, node.String())
+			}
+		}
+		if len(selectedNodes) == 0 {
+			return nil
 		}
-	}
-
-	if len(selectedNodes) == 0 {
-		return nil
-	}
-
-	// Save the image into a tar
-	dir, err := fs.TempDir("", "image-tar")
-	if err != nil {
-		return errors.Wrap(err, "failed to create tempdir")
-	}
-	defer os.RemoveAll(dir)
-	imageTarPath := filepath.Join(dir, "image.tar")
-
-	err = save(imageName, imageTarPath)
-	if err != nil {
-		return err
-	}
 
-	// Load the image on the selected nodes
-	fns := []func() error{}
-	for _, selectedNode := range selectedNodes {
-		selectedNode := selectedNode // capture loop variable
-		fns = append(fns, func() error {
-			return loadImage(imageTarPath, selectedNode)
-		})
+		// Load the image on the selected nodes
+		for _, selectedNode := range selectedNodes {
+			selectedNode := selectedNode // capture loop variable
+			imageName := imageName       // capture variable (for lint checks)
+			fns = append(fns, func() error {
+				// Save the image into a tar
+				dir, err := fs.TempDir("", "image-tar")
+				if err != nil {
+					return errors.Wrap(err, "failed to create tempdir")
+				}
+				defer os.RemoveAll(dir)
```

Comment:
i think we can re-use the same temp dir for all images.

Reply:
yes yes.. just noticed that few seconds ago. Was fixing that only.